subworkflows/local/contaminant_filter.nf

```diff
@@ -94,9 +146,30 @@ workflow CONTAMINANT_FILTER {
     ncrna_reads.set { pirna_reads }
 
     if (params.pirna) {
-        BLAT_PIRNA ( Channel.value( 'other' ), ch_reference_hairpin, ch_pirna )
-        ch_versions = ch_versions.mix(BLAT_PIRNA.out.versions)
-        INDEX_PIRNA ( BLAT_PIRNA.out.filtered_set )
+        // Search which hairpin miRNAs are present in the piRNA data
+        BLAT_PIRNA(ch_reference_hairpin, ch_pirna)
+
+        // Extract the significant hits
+        ch_program = Channel.of('BEGIN{FS="\t"}{if(\$11 < 1e-5) print \$2;}').collectFile(name:"program.txt")
```

Comment:
Placeholder for removing ch_program if definining it on top of the workflow outside of the if statements